src/key-store.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * The KeyStore in Tram-One is a basic key-value object
+ * that needs to be persisted in the globalSpace.
+ *
+ * Currently this is used with useStore and useGlobalStore to keep
+ * track of what stores need to be cleaned up when removing elements
+ */
```

Comment:
This is heavily inspired by the EffectStore.

This store keeps track of store keys (e.g. `"app[{}]/test[{}]"`, and how many observers they have (just a number). 

If the number of observers goes down to zero after removing an element from the DOM, then we remove the store.